client/lib/query-manager/key.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import sortBy from 'lodash/sortBy';
+import toPairs from 'lodash/toPairs';
+import fromPairs from 'lodash/fromPairs';
+
+/**
+ * QueryKey manages the serialization and deserialization of a query key for
+ * use in tracking query results in an instance of QueryManager.
+ */
+export default class QueryKey {
+	/**
+	 * Returns a serialized query, given a query object
+	 *
+	 * @param  {Object} query Query object
+	 * @return {String}       Serialized query
+	 */
+	static stringify( query ) {
+		return JSON.stringify( sortBy( toPairs( query ), ( pair ) => pair[ 0 ] ) );
```

Comment:
Would it be possible to use the following, or would it still be sometimes incorrect after a sort?

```javascript
return JSON.stringify( fromPairs( sortBy( toPairs( query ), ( pair ) => pair[ 0 ] ) ) );
```